mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -170,19 +176,66 @@ pub(super) async fn init_light_client(
 
     let tonic_channel = Channel::builder(lightwalletd_url)
         .tls_config(ClientTlsConfig::new())
-        .map_to_mm(ZcoinLightClientInitError::TlsConfigFailure)?
+        .map_to_mm(ZcoinClientInitError::TlsConfigFailure)?
         .connect()
         .await
-        .map_to_mm(ZcoinLightClientInitError::ConnectionFailure)?;
-    let grpc_client = CompactTxStreamerClient::new(tonic_channel);
+        .map_to_mm(ZcoinClientInitError::ConnectionFailure)?;
 
     let (sync_status_notifier, sync_watcher) = channel(1);
     let (on_tx_gen_notifier, on_tx_gen_watcher) = channel(1);
 
     let wallet_db = Arc::new(Mutex::new(wallet_db));
     let sync_handle = SaplingSyncLoopHandle {
         current_block: BlockHeight::from_u32(0),
-        grpc_client,
+        rpc_client: ZRpcClient::Light(CompactTxStreamerClient::new(tonic_channel)),
+        blocks_db,
+        wallet_db: wallet_db.clone(),
+        consensus_params,
+        sync_status_notifier,
+        on_tx_gen_watcher,
+        watch_for_tx: None,
+    };
+    let abort_handle = spawn_abortable(light_wallet_db_sync_loop(sync_handle));
+
+    Ok((
+        SaplingSyncConnector::new_mutex_wrapped(sync_watcher, on_tx_gen_notifier, abort_handle),
+        wallet_db,
+    ))
+}
+
+pub(super) async fn init_native_client(
+    native_client: NativeClient,
+    cache_db_path: PathBuf,
+    wallet_db_path: PathBuf,
+    consensus_params: ZcoinConsensusParams,
+    evk: ExtendedFullViewingKey,
+) -> Result<(AsyncMutex<SaplingSyncConnector>, WalletDbShared), MmError<ZcoinClientInitError>> {
+    let blocks_db =
+        async_blocking(|| BlockDb::for_path(cache_db_path).map_to_mm(ZcoinClientInitError::BlocksDbInitFailure))
+            .await?;
+
+    let wallet_db = async_blocking({
```

Comment:
Initializing block and wallet DB is the same for both native and light modes. Please remove code duplication.

Reply:
Made function `create_wallet_db`.
Done.